### PR TITLE
add crashpad

### PR DESCRIPTION
**Pull request by yamashi**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/c/crashpad/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package("crashpad")
+    set_homepage("https://chromium.googlesource.com/crashpad/crashpad/+/refs/heads/main/README.md")
+    set_description("Crashpad is a crash-reporting system.")
+
+    if is_host("windows") then
+        if is_arch("x64", "x86_64") then
+            set_urls("http://get.backtrace.io/crashpad/builds/crashpad-release-x86-64-$(version).zip")
+            add_versions("stable", "b3facf8a802dfd12daf4d9fba416f4d4b5df0ae544afa14080662fa978aa18cb")
+            add_versions("latest", "7705073dfff89c376303cacea3a6f8c63322f77566ad5cdbe37060cf3cef9f8b")
```
Comment (waruqi) [inline packages/c/crashpad/xmake.lua:9]: incorrect version, please use a fake version with commit date, like "2021.1.1" instead of `stable` and `latest`

Comment (waruqi) [reply packages/c/crashpad/xmake.lua:9]:
```lua
set_urls("http://get.backtrace.io/crashpad/builds/crashpad-release-x86-64-$(version).zip", function (version)
    local map = {
       ["2021.1.1"] = "stable",
       ...
    }
    return map[tostring(version)]
end)
```

Comment (yamashi) [reply packages/c/crashpad/xmake.lua]: I implemented the suggestion but am getting a weird error now

File: packages/c/crashpad/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            ["2021.8.1"] = "stable",
+            ["2022.4.16"] = "latest"
+        }
+    
+        function map_version(version)
+            return map[tostring(version)]
+        end
+    
+        if is_arch("x64", "x86_64") then
+            set_urls("http://get.backtrace.io/crashpad/builds/crashpad-release-x86-64-$(version).zip", map_version)
```
Comment (waruqi) [inline packages/c/crashpad/xmake.lua:17]: `{version = map_version}`

**Comment by sanqideshi** [comment]
can you write `on_test` function of crashpad , for windows platform?

**Pull request merged**